magento/magento2
# Refactoring for Customer\Controller\Address\FormPost

### Description (*)
Initially this should be fix for #5013 
But current implementation saving customer addresses by storefront is different then REST API or Adminhtml controller:  uses AddressRepository::save instead if CustomerRepository::save as aggregate root saving.

### Fixed Issues (if relevant)

1. #5013 
2. New found issue: 
During saving address with VAT & group changing implemented in https://github.com/arijit-de/magento2/blob/2.3-develop/app/code/Magento/Customer/Observer/AfterAddressSaveObserver.php observer was run code `$customer->save()` which have OLD address collection which loaded in memory before new address data persist to database. So customer address re-saved with old data on `$customer->save()` call.

### Manual testing scenarios (*)


    1. First pull a customer with using REST API and note down the updated time.
    2. Login to Frontend with the same account and edit the addresses.
    3. Pull the same again and see the updated time also got changed.


### Contribution checklist (*)
